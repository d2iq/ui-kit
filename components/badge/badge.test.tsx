import React from "react";
import renderer from 'react-test-renderer';
import Badge from '../badge';

const StringComponent = () : JSX.Element => {
  return (
    <span>string</span>
  )
};

describe("Badge", () => {
  it("default", () => {
    expect(renderer
    .create(<Badge>default</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("success", () => {
    expect(renderer
    .create(<Badge appearance="success">success</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("primary", () => {
    expect(renderer
    .create(<Badge appearance="primary">primary</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("danger", () => {
    expect(renderer
    .create(<Badge appearance="danger">danger</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("warning", () => {
    expect(renderer
    .create(<Badge appearance="warning">warning</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("outline", () => {
    expect(renderer
    .create(<Badge appearance="outline">outline</Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("accept jsx as children", () => {
    expect(renderer
    .create(<Badge><StringComponent /></Badge>)
    .toJSON()).toMatchSnapshot()
  });

  it("can have onClick event", () => {
    const foo = () => {
      return true;
    };

    expect(renderer
    .create(<Badge onClick={foo}>onClick badge</Badge>)
    .toJSON()).toMatchSnapshot()
  });
});
