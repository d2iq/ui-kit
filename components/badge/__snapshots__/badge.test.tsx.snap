// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge accept jsx as children 1`] = `
<span
  className="css-h6dpn5"
>
  <span>
    string
  </span>
</span>
`;

exports[`Badge can have onClick event 1`] = `
<span
  className="css-6jnl7a"
  onClick={[Function]}
>
  onClick badge
</span>
`;

exports[`Badge danger 1`] = `
<span
  className="css-1rtadzi"
>
  danger
</span>
`;

exports[`Badge default 1`] = `
<span
  className="css-h6dpn5"
>
  default
</span>
`;

exports[`Badge outline 1`] = `
<span
  className="css-1itz6ao"
>
  outline
</span>
`;

exports[`Badge primary 1`] = `
<span
  className="css-12gdqng"
>
  primary
</span>
`;

exports[`Badge success 1`] = `
<span
  className="css-k4847p"
>
  success
</span>
`;

exports[`Badge warning 1`] = `
<span
  className="css-1ufo5ae"
>
  warning
</span>
`;
